core/app/models/spree/order_updater.rb

```diff
@@ -164,5 +163,51 @@ def update_payment_state
     def round_money(n)
       (n * 100).round / 100.0
     end
+
+    def update_promotions
+      [*line_items, *shipments, order].each do |item|
+        promotion_adjustments = item.adjustments.select(&:promotion?)
+
+        promotion_adjustments.each(&:update!)
+        Spree::Config.promotion_chooser_class.new(promotion_adjustments).update
+        item.promo_total = promotion_adjustments.select(&:eligible?).sum(&:amount)
+      end
+    end
+
+    def update_taxes
+      [*line_items, *shipments].each do |item|
+        tax_adjustments = item.adjustments.select(&:tax?)
+
+        tax_adjustments.each(&:update!)
+        item.included_tax_total   = tax_adjustments.select(&:included?).sum(&:amount)
+        item.additional_tax_total = tax_adjustments.reject(&:included?).sum(&:amount)
+      end
+    end
+
+    def update_cancellations
+      [*line_items, *shipments].each do |item|
+        item.adjustments.select(&:cancellation?).each(&:update!)
+      end
+    end
+
+    def update_item_totals
+      [*line_items, *shipments].each do |item|
+        item_cancellation_total = item.adjustments.select(&:cancellation?).sum(&:amount)
```

Comment:
It feels weird that `promo_total`, `included_tax_total`, and `additional_tax_total` are summed in their respective methods, but cancellations are not.